### PR TITLE
Plane: remove support for alt wait

### PR DESCRIPTION
**Pull request by IamPete1**
This removes support for altitude wait mission command, used for high altitude balloon launches. Very rarely used functionality that can now be replicated with scripting.

### CODE REVIEW

**Comment by IamPete1** [comment]
saves ~500 bytes on plane and a handful on everything else.

**Review by tridge — CHANGES_REQUESTED** [review]
i wouldn't want to remove this until we have a working replacement lua script, perhaps with NAV_SCRIPT_TIME, with the same features
This feature is being actively used by balloon drop projects

**Comment by IamPete1** [comment]
This now adds the ability to run commands from scripting. It lays a frame works for more generic way-point support in the future. Behavior with new script is same as before. Only unexpected thing that might happen would be if one were to switch out of auto and back while the nav command is running.

File: libraries/AP_Mission/AP_Mission.cpp [hunk 1]
```diff
@@ -1627,6 +1622,13 @@ bool AP_Mission::mission_cmd_to_mavlink_int(const AP_Mission::Mission_Command& c
         packet.param4 = cmd.content.scripting.p3;
         break;
 
+    case MAV_CMD_NAV_ALTITUDE_WAIT:
```
Comment (tridge) [inline libraries/AP_Mission/AP_Mission.cpp:1625]: i think we need to keep the compatibility with the existing storage format

File: libraries/AP_Scripting/examples/Nav_Altitude_Wait.lua [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  update_servos()
+
+  loc = ahrs:get_location()
+  vel = ahrs:get_velocity_NED()
+  if (not loc) or (not vel) then
+    -- wait to get location and velocity
+    filtered_sink_rate = 0
+    return run_alt_wait, 100
+  end
+  loc:change_alt_frame(ABOVE_HOME)
```
Comment (tridge) [inline libraries/AP_Scripting/examples/Nav_Altitude_Wait.lua:105]: need to check if original handled absolute/terrain frames

**Review by tridge — CHANGES_REQUESTED** [review]
i need to do some testing on this

**Comment by IamPete1** [comment]
https://github.com/ArduPilot/ardupilot/pull/25391 achieves the core of what I was after, removing the odd early return. I do think generic scripting waypoints is a good idea in the future.

**Pull request closed without merging**